app/src/main/java/ch/epfl/polybazaar/database/Datastore.java

```diff
@@ -0,0 +1,20 @@
+package ch.epfl.polybazaar.database;
+
+import androidx.annotation.NonNull;
+import ch.epfl.polybazaar.database.callback.SuccessCallback;
+
+
+public interface Datastore {
+
```

Comment:
We cannot leave out getAllDataInCollection() without consequences, that method is used in other parts of the app... But since mocking QuerySnapshot would be cumbersome, maybe we can figure out a way to not need this method any more.